File: myapp_new/lib/features/driver/profile/models/user_profile.dart
class UserProfile {
  int? userId;
  String? fullName;
  String? phone; 
  String? avatarUrl;
  String? defaultAddress;
  String? idCardImageUrl;
  String? driverLicenseUrl;
  String? vehicleInfo;
  int? rewardPoints;
  String? tier;

  int? totalOrders; 

  UserProfile({
    this.userId,
    this.fullName,
    this.phone, 
    this.avatarUrl,
    this.defaultAddress,
    this.idCardImageUrl,
    this.driverLicenseUrl,
    this.vehicleInfo,
    this.rewardPoints,
    this.tier, 
  });

  factory UserProfile.fromJson(Map<String, dynamic> json) {
    return UserProfile(
      userId: json['userId'],
      fullName: json['fullName'],
      phone: json['phone'], 
      avatarUrl: json['avatarUrl'],
      defaultAddress: json['defaultAddress'],
      idCardImageUrl: json['idCardImageUrl'],
      driverLicenseUrl: json['driverLicenseUrl'],
      vehicleInfo: json['vehicleInfo'],
      rewardPoints: json['rewardPoints'] ?? 0,
      tier: json['tier'], 
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'userId': userId,
      'fullName': fullName,
      'phone': phone,
      'avatarUrl': avatarUrl,
      'defaultAddress': defaultAddress,
      'idCardImageUrl': idCardImageUrl,
      'driverLicenseUrl': driverLicenseUrl,
      'vehicleInfo': vehicleInfo,
      'rewardPoints': rewardPoints,
      'tier': tier, 
    };
  }
}

File: SmartShip_backend/user-service/src/main/java/com/smartship/userservice/entity/UserProfile.java
package com.smartship.userservice.entity;

import com.fasterxml.jackson.annotation.JsonProperty;
import jakarta.persistence.*; // Sử dụng cho Spring Boot 3.x
import lombok.Getter;
import lombok.Setter;
import org.hibernate.annotations.JdbcTypeCode;
import org.hibernate.type.SqlTypes;

@Entity
@Table(name = "user_profiles")
@Getter
@Setter
public class UserProfile {
    @Id
    @Column(name = "user_id")
    private Long userId;

    // UserProfile.java
    @Column(name = "full_name")
    private String fullName;

    @Column(name = "phone")
    private String phone;

    private String avatarUrl;
    private String defaultAddress;
    private String idCardImageUrl;
    private String driverLicenseUrl;

    @Column(name = "id_card_back_url")
    private String idCardBackUrl;

    @Column(name = "vehicle_reg_url")
    private String vehicleRegUrl;

    @JdbcTypeCode(SqlTypes.JSON)
    @Column(columnDefinition = "jsonb") // Lưu thông tin xe dạng JSON [cite: 101, 175]
    private String vehicleInfo;

    @Column(name = "reward_points")
    private Integer rewardPoints = 0; // Điểm thưởng mặc định là 0 [cite: 102, 176]

    @Column(name = "total_orders")
    private Integer totalOrders = 0;

    @Column(name = "tier")
    private String tier = "BRONZE";

    @Column(name = "status")
    private String status = "ACTIVE";
}